### PR TITLE
OD-12994: DFP ad unit code

### PR DESCRIPTION
**Pull request by Konstantseven**
Added field for DFP ad unit code and GPID propagation

### CODE REVIEW

File: src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto [hunk 1]
```diff
@@ -150,6 +150,9 @@ message ImpExt {
     // Supported viewability vendors. For list of supported vendors, visit:
     // https://resources.rubiconproject.com/resource/publisher-resources/xapi-specifications/#4.9.
     repeated string viewabilityvendors = 3;
+
+    // Field for DFP ad unit code and GPID propagation.
```
Comment (shkil) [inline src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto:154]: DFP ad unit code is treated in a special way for Magnite only. I would leave a simple comment: "Ad unit code"

Comment (Konstantseven) [reply src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto]: Thanks, fixed

File: src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto [hunk 2]
```diff
@@ -150,6 +150,9 @@ message ImpExt {
     // Supported viewability vendors. For list of supported vendors, visit:
     // https://resources.rubiconproject.com/resource/publisher-resources/xapi-specifications/#4.9.
     repeated string viewabilityvendors = 3;
+
+    // Ad unit code.
```
Comment (simontrasler) [inline src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto:154]: Please update the comment to something like: Global Placement ID, as defined by TheTradeDesk. The goal is that all participants in buying impressions for the same placement get the same identifier, and that the identifier (as much as possible) is unique for the placement. (In general, this field will be populated with one of the DFP Ad Unit Code, Prebid Ad Slot, or some value provided by the publisher or ad server.)

File: src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto [hunk 3]
```diff
@@ -150,6 +150,12 @@ message ImpExt {
     // Supported viewability vendors. For list of supported vendors, visit:
     // https://resources.rubiconproject.com/resource/publisher-resources/xapi-specifications/#4.9.
     repeated string viewabilityvendors = 3;
+
+    // Global Placement ID. The unique identifier of the placement, that is sent
```
Comment (eskenny) [inline src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto:154]: Should be no comma after "placement"
> the placement that is sent

File: src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto [hunk 4]
```diff
@@ -150,6 +150,12 @@ message ImpExt {
     // Supported viewability vendors. For list of supported vendors, visit:
     // https://resources.rubiconproject.com/resource/publisher-resources/xapi-specifications/#4.9.
     repeated string viewabilityvendors = 3;
+
+    // Global Placement ID. The unique identifier of the placement, that is sent 
+    // to all participants in buying impressions for this placement (in general,
```
Comment (eskenny) [inline src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto:155]: Parenthesis should be the start of a new sentence
> for this placement. (In general,

File: src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto [hunk 5]
```diff
@@ -150,6 +150,12 @@ message ImpExt {
...
+    // this field will be populated with one of the DFP Ad Unit Code, Prebid Ad Slot,
+    // or value, provided by the publisher or ad server).
```
Comment (eskenny) [inline src/proto/com/magnite/openrtb/v2/openrtb-xapi.proto:157]: Should be no comma after value
> // or value provided by

**Review by eskenny — COMMENTED** [review]
A few grammatical comments. I'll defer to others for the actual contents, but should we mention TTD?

**Review by eskenny — APPROVED** [review]

**Pull request merged**